src/fsharp/lexhelp.fs

```diff
@@ -335,14 +337,18 @@ module Keywords =
             match s with 
             | "__SOURCE_DIRECTORY__" ->
```

Comment:
Yes. This is guarding against cases where Roslyn APIs don't provide the document name (like colorizing a particular span).